js/src/dom/data.js

```diff
@@ -21,7 +21,7 @@ export default {
 
     // make it clear we only want one instance per element
     // can be removed later when multiple key/instances are fine to be used
-    if (!instanceMap.has(key) && instanceMap.size !== 0) {
+    if (instanceMap.has(key) && instanceMap.size !== 0) {
```

Comment:
Suppose it will work with this too. However, the Pr will need a test too
```suggestion
    if (instanceMap.size !== 0) {
```

Reply:
OK. I will.